WIP: histogram rework additions and review

kahaaga opened the pull request:
I'm going through all points raised in #112 here. This will need a quick review before merging with #112 if anything major shows up - I'll give notice when I'm done with my part.

Todos:

- [x] I had no idea that the source code had marginal, join, and this weird 5th way to specify ε in the binning. I know nothing about them. Are they used in the transfer operator? If so, all this stuff needs to be moved into other files. I have no idea if they work with my changes. (of course, if these things are for the transfer operator, they should be modified in a different PR)

Some code was old, some still is used with the transfer operator. I've moved the relevant code that still is used, and deleted the rest.


- [x] > **All of this infastructure of marginal histogram visits, I had no idea about, and it wasn't really documented anywhere... Is this part of the VisitationFrequency stuff? That's what I'm working on in this PR.**  

This seems to all be remnants of ancient code, and is nowhere to be found in upstream packages. For the transfer operator, it has been superseded by the GroupSlices functionality.

- [x] encode_box_visits.jl file must be deleted or moved to where it is actually used. It is not used in current VisitationFrequency code.

See comment above.  Conclusion: simply delete it.

kahaaga commented on src/probabilities_estimators/histograms/rectangular_binning.jl:
```diff
@@ -0,0 +1,172 @@
+function encode_as_bins(x::Vector_or_Dataset, b::RectangularBinEncoder)
+    (; mini, edgelengths) = b
+    # Map each datapoint to its bin edge (hence, we are symbolizing the x here)
+    # (notice that this also works for vector x, and broadcasting is ignored)
```
What does "this also works for vector x" mean, precisely? Is the point that encoding/symbolization also works for both `Vector` and `Dataset` input? If so, I don't think we need to comment that explicitly, because it is implied by the type annotation on the `x` argument (`Vector_or_Dataset`).

kahaaga commented on src/probabilities_estimators/histograms/rectangular_binning.jl:
```diff
@@ -0,0 +1,172 @@
+end
+
+# This function is the same as `symbolize`.
+function encode_as_bins(x::Vector_or_Dataset, b::RectangularBinEncoder)
```
Do we need the `encode_as_bins`? Can't we just use `symbolize(x, b)` directly?

Datseris commented:
cool, just tag me when you want a review!

Datseris commented:
Are we addressing the comments you raised on #112 here, or are we merging this in, and then I address those comments in #112 ?

kahaaga commented:
> Are we addressing the comments you raised on https://github.com/JuliaDynamics/Entropies.jl/pull/112 here, or are we merging this in, and then I address those comments in https://github.com/JuliaDynamics/Entropies.jl/pull/112 ?

Sorry, I started commenting while having checked out the wrong PR. I'm fixing stuff here, then merging this with #112  once I'm done. I also commented on the original PR, so just address the comments there!

kahaaga commented:
Ah, this has `main` as the target branch for the merge. Will close this and open a new PR to #112.
